quail/processes/wps_climdexInput_raw.py

```diff
@@ -0,0 +1,340 @@
+import os
+from rpy2 import robjects
+from pywps import Process, LiteralInput, ComplexInput, ComplexOutput, Format
+from pywps.app.Common import Metadata
+from tempfile import NamedTemporaryFile
+
+from wps_tools.utils import log_handler, collect_args, common_status_percentages
+from wps_tools.io import log_level
+from quail.utils import get_package, logger, load_rdata_to_python, save_python_to_rdata
+from quail.io import (
+    tmax_column,
+    tmin_column,
+    prec_column,
+    tavg_column,
+    base_range,
+    cal,
+    date_fields,
+    date_format,
+    n,
+    northern_hemisphere,
+    quantiles,
+    temp_qtiles,
+    prec_qtiles,
+    max_missing_days,
+    min_base_data_fraction_present,
+    output_file,
+    vector_name,
+)
+
+
+class ClimdexInputRaw(Process):
+    """
+    Process for creating climdexInput object from data already ingested into R
+    """
+
+    def __init__(self):
+        self.status_percentage_steps = dict(
+            common_status_percentages,
+            **{
+                "prepare_params": 10,
+                "save_rdata": 90,
+            },
+        )
+        inputs = [
+            ComplexInput(
+                "tmax_file",
+                "daily maximum temperature data file",
+                abstract="Name of file containing daily maximum temperature data.",
+                min_occurs=0,
+                max_occurs=1,
+                supported_formats=[
+                    Format("application/x-gzip", extension=".rda", encoding="base64"),
+                ],
+            ),
+            ComplexInput(
+                "tmin_file",
+                "daily minimum temperature data file",
+                abstract="Name of file containing daily minimum temperature data.",
+                min_occurs=0,
+                max_occurs=1,
+                supported_formats=[
+                    Format("application/x-gzip", extension=".rda", encoding="base64"),
+                ],
+            ),
+            ComplexInput(
+                "prec_file",
+                "daily total precipitation data file",
+                abstract="Name of file containing daily total precipitation data.",
+                min_occurs=1,
+                max_occurs=1,
+                supported_formats=[
+                    Format("application/x-gzip", extension=".rda", encoding="base64"),
+                ],
+            ),
+            ComplexInput(
+                "tavg_file",
+                "mean temperature data file",
+                abstract="Name of file containing daily mean temperature data.",
+                min_occurs=0,
+                max_occurs=1,
+                supported_formats=[
+                    Format("application/x-gzip", extension=".rda", encoding="base64"),
+                ],
```

Comment:
Is there a reason you don't want to store all the data in the same rda file?

Reply:
I just wanted the process to be flexible with the inputs. There may be cases that a user wants to extract `tmax` from one file and `tmin` from another.